crane/container.go

```diff
@@ -639,16 +639,29 @@ func (c *container) Logs(follow bool, tail string) (stdout, stderr io.Reader) {
 }
 
 // Push container
-func (c *container) Push() {
-	if len(c.Image()) > 0 {
-		fmt.Printf("Pushing image %s ... ", c.Image())
-		args := []string{"push", c.Image()}
+func (c *container) Push(tag string) {
+	imageName := c.getTaggedImageName(tag)
+	if len(imageName) > 0 {
+		fmt.Printf("Pushing image %s ... ", imageName)
+		args := []string{"push", imageName}
 		executeCommand("docker", args)
 	} else {
 		print.Noticef("Skipping %s as it does not have an image name.\n", c.Name())
 	}
 }
 
+func (c *container) getTaggedImageName(tag string) {
+	imageName := c.Image()
+	if len(imageName) > 0 {
+		if len(tag) > 0 {
+			imageParts := strings.Split(imageName, ":")
+			imageParts[1] = tag
```

Comment:
Maybe I am missing something, but I just wanted to point out that this look like it will panic if the result of the split is a single-item array.